### ISSUE TITLE
Remove "eager" usage of settings

### ISSUE THREAD

**Issue by tomage** [jyveapp/django-action-framework#3]
In `daf.rest_framework`, there's a defintion of an exception, `APIException`, which sets `status_code` based on a settings variable.

This hasn't posed issues before when we've imported `daf.rest_framework` whenever we want to define DRF actions.

But recently, I tried to import `daf.rest_framework` in a python file much "closer" to the manage.py script, and found that it would complain - namely because Django settings were being accessed before they were loaded.

I figured-whilst probably not a big issue per se-that it'd still be prudent to try and access these settings a bit later on, for example by using a property, like so:

```python
class APIException(drf_exceptions.APIException):
    """
    The base error class raised by `raise_drf_error`.
    """

    @property
    def status_code(self):
        return getattr(
            settings,
            'DAF_DEFAULT_REST_FRAMEWORK_ERROR_STATUS_CODE',
            drf_status.HTTP_400_BAD_REQUEST,
        )

    default_detail = 'Invalid input.'
    default_code = 'invalid'
```

This has some nice properties:

- One can dynamically set/change `settings.DAF_DEFAULT_REST_FRAMEWORK_ERROR_STATUS_CODE`, and have the default status code of the exception change too.
- One can still override it on the APIException class
- One can still subclass APIException, and override the status_code

To corroborate, I found this tidbit in Django docs:
> [...example importing and using settings in top-level...] That means that the ability for third parties to import the module at the top level is incompatible with the ability to configure the settings object manually, or makes it very difficult in some circumstances.

Cf. https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#use-of-django-conf-settings

What do you think @wesleykendall ?

**Comment by wesleykendall** [comment]
Sounds good to me @tomage !

**Comment by tomage** [comment]
Fixed via #4.
